### PR TITLE
BXMSPROD-685: Weld 3.x upgrade

### PR DESCRIPTION
**Pull request by mareknovotny**
https://issues.redhat.com/browse/BXMSPROD-685

Merge with these PR together:

- https://github.com/kiegroup/appformer/pull/969
- https://github.com/kiegroup/drools/pull/3077
- https://github.com/kiegroup/jbpm/pull/1654
- https://github.com/kiegroup/kie-wb-common/pull/3409
- https://github.com/kiegroup/drools-wb/pull/1373
- https://github.com/kiegroup/jbpm-wb/pull/1434
- https://github.com/kiegroup/kie-wb-distributions/pull/1055

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -179,9 +179,9 @@
     <version.org.glassfish.jaxb>2.3.2</version.org.glassfish.jaxb>
     <version.org.gwtbootstrap3>1.0.1</version.org.gwtbootstrap3>
     <version.org.gwtbootstrap3-extras>1.0.2</version.org.gwtbootstrap3-extras>
-    <version.org.hibernate>5.3.7.Final</version.org.hibernate>
+    <version.org.hibernate>5.3.14.Final</version.org.hibernate>
```
Comment (yesamer) [inline pom.xml:182]: @mareknovotny Very very minor question: I checked `WildFly18` BOM and the hibernate version is `5.3.13`. Is this intentional? Probably there is no problem to use `5.4.13`, just want to report it.
https://github.com/wildfly/wildfly/blob/26689e876b7c5d271e3d023539fa57c52bad503d/pom.xml#L334

Comment (mareknovotny) [reply pom.xml:182]: @yesamer this is more inline with what EAP 7.3.0 will bring

Comment (yesamer) [reply pom.xml:182]: @mareknovotny Thank you!

Comment (elguardian) [reply pom.xml:182]: @yesamer @mareknovotny  hiberante 5.4.14 version will rise problems in sql scripting and validation. better be done that bump by eng than prod team.

Comment (mareknovotny) [reply pom.xml:182]: @elguardian it is upgrade to 5.3.14 not 5.4.14 ;) so the upgrade is on micro version only.

Comment (elguardian) [reply pom.xml:182]: @mareknovotny I was answering @yesamer bout bumping to 5.14... there is one problem.

**Comment by mareknovotny** [comment]
So @ederign @yesamer @adrielparedes I need quickly a help with tests in Appformer and drools-wb where upgraded Weld shows 
`java.lang.IncompatibleClassChangeError: javax.enterprise.inject.Default and javax.enterprise.inject.Default$Literal disagree on InnerClasses attribute`

File: pom.xml (hunk 1 above)
Comment (yesamer) [reply pom.xml:182]: @elguardian Yes, I wrongly wrote `5.4.14` version in my first comment, which was a type ( I meant `5.3.14`). My apologies for the typo and thank you for your clarification.

**Comment by ederign** [comment]
@mareknovotny I'll take a look at this, this week.

**Comment by mareknovotny** [comment]
there is errai PR https://github.com/errai/errai/pull/388 fyi @ederign

**Comment by mareknovotny** [comment]
Jenkins execute full downstream build

**Comment by mareknovotny** [comment]
Jenkins execute full downstream build

**Comment by mareknovotny** [comment]
Jenkins do fdb

**Comment by mareknovotny** [comment]
So i did here update to Errai 4.10.0-SNAPSHOT and Weld core 3.1.5..SP1 to check if everything works as expected. Before merging we need to release Errai final version and update it here again

**Comment by mareknovotny** [comment]
Jenkins do fdb

**Comment by mareknovotny** [comment]
Jenkins do fdb

**Comment by mareknovotny** [comment]
Jenkins do fdb

**Comment by mareknovotny** [comment]
Jenkins do fdb

**Comment by mareknovotny** [comment]
Jenkins do fdb

**Review by yesamer — APPROVED** [review]

**Pull request merged**